### PR TITLE
implemented a rule to detect overwriting parent scope vars in for loo…

### LINKED ISSUE #98: Report for loops overwriting variables from parent scope (a la foreach)
https://phpstan.org/r/8f361ed5-7e15-4ffc-a6b0-3b5c33a3b419

PhpStorm complains about code like this, because an argument is being overwritten by a loop, which might not be intended behaviour.

I found it strange that overwriting from `foreach` context is reported, but not from `for` context.

Is there a reason for this not being reported, or just a missing feature?

### PR DESCRIPTION
**Pull request by dktapps**
…p init

closes #98

### CODE REVIEW

**Comment by dktapps** [comment]
Updated, I hope this is suitable now 🙂

File: rules.neon [hunk 1]
```diff
@@ -22,6 +22,7 @@ rules:
 	- PHPStan\Rules\DisallowedConstructs\DisallowedImplicitArrayCreationRule
 	- PHPStan\Rules\DisallowedConstructs\DisallowedShortTernaryRule
 	- PHPStan\Rules\ForeachLoop\OverwriteVariablesWithForeachRule
+	- PHPStan\Rules\ForLoop\OverwriteVariablesWithForLoopInitRule
```
Comment (ondrejmirtes) [inline rules.neon:25]: Adding rule to `rules` automatically applies the rule tag, this needs to be added as a tag-less service in the `services` section.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Otherwise ready to merge 👍

**Comment by ondrejmirtes** [comment]
Thank you!

**Comment by ondrejmirtes** [comment]
Tagged as 0.12.5.

**Comment by dktapps** [comment]
Awesome, thanks!

**Pull request merged**